Autor: Jedrzej Szadejko
CTR52-CPP Guarantee that library functions do not overflow
Kopiowanie danych do zmiennej ktora nie jest wystarczajaca aby je pomiescic powoduje przepelnienie bufora
diff --git a/CTR52-CPP/main.cpp b/CTR52-CPP/main.cpp
index 62b595f..16e56be 100644
--- a/CTR52-CPP/main.cpp
+++ b/CTR52-CPP/main.cpp
@@ -2,13 +2,13 @@
 
 void f(const std::vector<int>& src)
 {
-	std::vector<int> dest;
+	std::vector<int> dest(src.size());
 	std::copy(src.begin(), src.end(), dest.begin());
 }
 
 void g()
 {
-	std::vector<int> v;
+	std::vector<int> v(10);
 	std::fill_n(v.begin(), 10, 0x42);
 }
 
@@ -21,4 +21,4 @@ int main()
 	g();
 	
 	return 0;
-}
\ No newline at end of file
+}
